lit/docs/jobs/steps/set_pipeline.lit

```diff
@@ -0,0 +1,125 @@
+\use-plugin{concourse-docs}
+
+\title{\code{set_pipeline} step}{set-pipeline-step}
+\warn{The \code{set_pipeline} step was introduced in Concourse v5.8.0. It is
+considered an \bold{experimental} feature, until its associated
+\link{RFC}{https://github.com/concourse/rfcs/pull/31} is resolved.}
+
+Configures a Concourse pipeline.
+
+\right-side{Examples}{
+  \example{Self-reconfiguring pipeline}{
+    This is a reasonable way to get Concourse to mimic the default behaviour of
+    some other CI systems, where the automation configuration is automatically
+    read from a source code repository and updated on every commit.
```

Comment:
...However it's probably worth documenting these nuances and painting a picture of the future we're striving for, since people might try to use it in this way regardless. This kind of content can probably appear in the main documentation page. I can work on that if you like!